Functions/Statistics/xASL_stat_PrintStats.m

```diff
@@ -191,19 +211,22 @@
                         SessionColumn = find(strcmpi(x.S.SetsName, 'session'));
                         if isempty(SessionColumn) || length(SessionColumn)>1
                             fprintf(2, 'Could not find session data\n');
-                        else
-                            SessionN = iSubjSess;
-                            if isempty(SessionN) || ~isnumeric(SessionN)
-                                fprintf(2, ['Something wrong with session ' SessionID '\n']);
+						else
+                            if isempty(iSubjSess) || ~isnumeric(iSubjSess)
+                                fprintf(2, ['Something went wrong with ' SubjectID '_' SessionID '\n']);
                             elseif SessionIndex>x.dataset.nSessions
-                                if ~max(printedSessionN==SessionN)
+                                if ~max(printedSessionN==iSubjSess)
                                     fprintf(2, 'Could not find values for other covariates\n');
                                 end
-                                printedSessionN = [printedSessionN SessionN];
+                                printedSessionN = [printedSessionN iSubjSess];
                             else
                                 
                                 %% Print the covariates and data
-                                iSubjectSession_SetsID = x.dataset.nSessions*(SubjectIndex-1)+SessionN;
+                                % SetsID -> contains throughout ExploreASL the stats read from participants_id, e.g., subject, session, covariates, names, etc
+                                % iSubjectSession_SetsID here defines the names that will be printed
+                                % _DAT & _DATA is about the data (mostly numerical), the stats coming out of e.g., xASL_wrp_GetROIStatistics that will
+                                % be printed
+                                iSubjectSession_SetsID = x.dataset.nSessions*(SubjectIndex-1)+SessionIndex;
```

Comment:
Check lines 132-138 - it's a bit worrisome that these differ from what is here. Also the comment on line 132 isn't very reassuring. I propose to adapt lines 132-138 to match what we have here and what I've tested.

Any better idea on this?

Reply:
Michael Stritt added that non-reassuring comment 3 years ago :)